### PR TITLE
PR: recursive goal parsing and later call of make_constraints

### PR DESCRIPTION
**Pull request by K3cks**
This should parse goals recursively to ensure that nested messages can still be parsed.
Also moved the call of make_constraints after sub_goals call.
The plugin `clean up goals` was moved to the end of the execution subtree

### CODE REVIEW

File: src/giskardpy/python_interface.py [hunk 1]
```diff
@@ -480,12 +480,28 @@ def set_json_goal(self,
...
+            else:
+                kwargs[k] = self.parse_messages(v)
         kwargs = replace_prefix_name_with_str(kwargs)
         constraint.parameter_value_pair = json.dumps(kwargs)
         self.cmd_seq[-1].constraints.append(constraint)
 
+    def parse_messages(self,
+                       val):
+
+        if isinstance(val, List):
```
Comment (ichumuh) [inline src/giskardpy/python_interface.py:492]: things for the typing module are not meant for `isinstance` checks, use `list` and so on

File: src/giskardpy/python_interface.py [hunk 2]
```diff
@@ -480,12 +480,28 @@ def set_json_goal(self,
...
                 del kwargs[k]
-            if isinstance(v, Message):
-                kwargs[k] = convert_ros_message_to_dictionary(v)
+            else:
+                kwargs[k] = self.parse_messages(v)
         kwargs = replace_prefix_name_with_str(kwargs)
         constraint.parameter_value_pair = json.dumps(kwargs)
         self.cmd_seq[-1].constraints.append(constraint)
 
+    def parse_messages(self,
```
Comment (ichumuh) [inline src/giskardpy/python_interface.py:489]: why is this a `self` function? from what I can tell, it doesn't need self. You should probably modify `convert_ros_message_to_dictionary` instead

File: src/giskardpy/tree/behaviors/ros_msg_to_goal.py [hunk 3]
```diff
@@ -58,7 +58,10 @@ def update(self):
             # traceback.print_exc()
             return Status.SUCCESS
         if self.god_map.get_data(identifier.collision_checker) != CollisionCheckerLib.none:
-            self.parse_collision_entries(move_cmd.collisions)
+            try:
+                self.parse_collision_entries(move_cmd.collisions)
+            except ConstraintInitalizationException:
```
Comment (ichumuh) [inline src/giskardpy/tree/behaviors/ros_msg_to_goal.py:63]: why are you catching this exception?

**Review by ichumuh — CHANGES_REQUESTED** [review]

File: src/giskardpy/tree/behaviors/ros_msg_to_goal.py (hunk 3 above)
Comment (K3cks) [reply src/giskardpy/tree/behaviors/ros_msg_to_goal.py]: I removed this now as this was not an intended to be pushed

File: src/giskardpy/python_interface.py (hunk 2 above)
Comment (K3cks) [reply src/giskardpy/python_interface.py]: yes correct, this functionality is now moved into `convert_ros_message_to_dictionary`

**Review by ichumuh — APPROVED** [review]

**Pull request merged**